cmd/docker/completions.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/context/store"
+	"github.com/spf13/cobra"
+)
+
+func registerCompletionFuncForGlobalFlags(dockerCli *command.DockerCli, cmd *cobra.Command) {
+	cmd.RegisterFlagCompletionFunc(
+		"context",
+		func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			names, err := store.Names(dockerCli.ContextStore())
+			if err != nil {
+				return nil, cobra.ShellCompDirectiveError
+			}
+
+			return filterForCompletion(names, toComplete), cobra.ShellCompDirectiveDefault
```

Comment:
If there are no context that match what the user has typed, do you want the shell to suggest file name?
If not, you should return `cobra.ShellCompDirectiveNoFileComp`.  For example:
```
$ docker --context D[tab][tab]  # There is a context starting with 'D'
Development
$ docker --context Do[tab][tab] # There is NO context starting with 'Do' so we would get file completion
Documents/   Dockerfile    Dockerfile.new   Dockerfile.ori
```